### PR TITLE
revised rdl comments

### LINKED ISSUE #470: Revise crypto core register descriptions in RDL
Sha256/sha512/HMAC/ECC register descriptions should be updated in RDL (and ultimately in the generated HTML). There are two issues I see:

1. Address offsets for each reg are hardcoded in the text description. These are auto-generated when we create the HTML doc anyway, and subject to becoming outdated, so they shouldn't be hardcoded in the text.
2. Bit descriptions are redundantly provided in the register text description, but also in the subsequent section relative to each register field. As a result, there are some mismatches, and some of the text descriptions are missing certain bit fields. E.g. [SHA256_CTRL](https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.sha256_reg.SHA256_CTRL) has a text description that is missing all mention of WNTZ-related bits. All reg "field descriptions" should be updated and then individual field descriptions should be removed from the global register "description" field.

### PR DESCRIPTION
**Pull request by mojtaba-bisheh**
resolved #470

### CODE REVIEW

**Review by upadhyayulakiran — APPROVED** [review]

File: src/ecc/rtl/ecc_reg.rdl [hunk 1]
```diff
@@ -38,8 +38,7 @@ addrmap ecc_reg {
     reg {
         name = "ECC component name register type definition";
         desc = "Two 32-bit read-only registers repereseting of the name
```
Comment (calebofearth) [inline src/ecc/rtl/ecc_reg.rdl:40]: nit: 'repereseting' is misspelled

File: src/ecc/rtl/ecc_reg.rdl [hunk 2]
```diff
@@ -53,8 +52,7 @@ addrmap ecc_reg {
     reg {
         name = "ECC component version register type definition";
         desc = "Two 32-bit read-only registers repereseting of the version
```
Comment (calebofearth) [inline src/ecc/rtl/ecc_reg.rdl:54]: nit: 'repereseting'

File: src/ecc/rtl/ecc_reg.rdl [hunk 3]
```diff
@@ -66,54 +64,46 @@ addrmap ecc_reg {
...
     reg {
-        name = "ECC component seed register type definition";
-        desc = "12 32-bit registers storing the 384-bit seed for keygen.
-                The seed can be any 384-bit value in [0 : 2^384-1].
-                These registers are located at ECC_base_address + 
-                0x0000_0080 to 0x0000_00AC in big-endian representation.";
+        name = "ECC SEED";
+        desc = "ECC component seed register type definition
+                12 32-bit registers storing the 384-bit seed for keygen.
+                The seed can be any 384-bit value in [0 : 2^384-1].";
```
Comment (calebofearth) [inline src/ecc/rtl/ecc_reg.rdl:106]: Probably still want to have 'big-endian representation' defined here.

File: src/hmac/rtl/hmac_reg.rdl [hunk 4]
```diff
@@ -64,54 +62,45 @@ addrmap hmac_reg {
...
         default resetsignal = reset_b;
-        field {desc = "Control init command bit"; singlepulse;} INIT = 1'b0;
-        field {desc = "Control next command bit"; singlepulse;} NEXT = 1'b0;
-        field {desc = "Zeroize all internal registers"; singlepulse;} ZEROIZE = 1'b0;
+        field {desc = "Control init command bit: Trigs the HMAC384 core to start the 
+                                processing for the key and the first padded 
+                                message block."; singlepulse;} INIT = 1'b0;
+        field {desc = "Control next command bit: ​Trigs the HMAC384 core to start the 
+                                processing for the remining padded message block. "; singlepulse;} NEXT = 1'b0;
+        field {desc = "Zeroize all internal registers: Zeroize all internal registers after HMAC process, to avoid SCA leakage."; singlepulse;} ZEROIZE = 1'b0;
```
Comment (calebofearth) [inline src/hmac/rtl/hmac_reg.rdl:76]: Add this back? "After each software write, hardware will erase the register."

File: src/ecc/rtl/ecc_reg.rdl (hunk 1 above)
Comment (mojtaba-bisheh) [reply src/ecc/rtl/ecc_reg.rdl]: Fixed

File: src/ecc/rtl/ecc_reg.rdl (hunk 2 above)
Comment (mojtaba-bisheh) [reply src/ecc/rtl/ecc_reg.rdl]: fixed!

File: src/ecc/rtl/ecc_reg.rdl (hunk 3 above)
Comment (mojtaba-bisheh) [reply src/ecc/rtl/ecc_reg.rdl:109]: added here and all other fields where requried.

File: src/hmac/rtl/hmac_reg.rdl (hunk 4 above)
Comment (mojtaba-bisheh) [reply src/hmac/rtl/hmac_reg.rdl]: singlepulse definition is added instead of this.

File: src/hmac/rtl/hmac_reg.rdl [hunk 5]
```diff
@@ -36,8 +36,7 @@ addrmap hmac_reg {
     reg {
         name = "HMAC384 component name register type definition";
         desc = "Two 32-bit read-only registers repereseting of the name
```
Comment (calebofearth) [inline src/hmac/rtl/hmac_reg.rdl:38]: Can you do a search and try to replace all instances of this typo?

File: src/hmac/rtl/hmac_reg.rdl [hunk 6]
```diff
@@ -157,10 +150,9 @@ addrmap hmac_reg {
 
     /* ---- HMAC384 Component Input LFSR Seed ---- */
     reg {
-        name = "HMAC384 component lfsr seed register type definition";
-        desc = "12 32-bit registers storing the 384-bit lfsr seed input.
-                These registers are located at HMAC384_base_address + 
-                0x0000_0130 to 0x0000_015C in big-endian representation.";
+        name = "HMAC384 LFSR SEED";
+        desc = "HMAC384 component lfsr seed register type definition
+                12 32-bit registers storing the 384-bit lfsr seed input.";
```
Comment (calebofearth) [inline src/hmac/rtl/hmac_reg.rdl:155]: big-endian

File: src/ecc/rtl/ecc_reg.rdl [hunk 7]
```diff
@@ -307,13 +291,12 @@ addrmap ecc_reg {
...
-        desc = "12 32-bit registers storing the 384-bit IV required 
+        name = "ECC IV";
+        desc = "ECC component IV register type definition
+                12 32-bit registers storing the 384-bit IV required 
                 for SCA countermeasures to randomize the inputs with no change
                 on the ECC outputs. 
-                The IV can be any 384-bit value in [0 : 2^384-1].
-                These registers are located at ECC_base_address + 
-                0x0000_0480 to 0x0000_04AC in big-endian representation.";
+                The IV can be any 384-bit value in [0 : 2^384-1].";
```
Comment (calebofearth) [inline src/ecc/rtl/ecc_reg.rdl:299]: big-endian

Comment (mojtaba-bisheh) [reply src/ecc/rtl/ecc_reg.rdl:299]: no need for IV since it's a random value

File: src/hmac/rtl/hmac_reg.rdl (hunk 6 above)
Comment (mojtaba-bisheh) [reply src/hmac/rtl/hmac_reg.rdl:155]: no need for lfsr_seed since it's a random value

**Review by calebofearth — APPROVED** [review]

**Review by upadhyayulakiran — APPROVED** [review]

**Pull request merged**